src/components/minor-button/minor-button.spec.tsx

```diff
@@ -0,0 +1,100 @@
+import React from "react";
+import TestRenderer from "react-test-renderer";
+import { shallow, mount } from "enzyme";
+import {
+  ButtonProps,
+  ButtonTypes,
+  SizeOptions,
+} from "../button/button.component";
+import MinorButton from "./minor-button.component";
+import { assertStyleMatch } from "../../__spec_helper__/test-utils";
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const render = (props: ButtonProps, renderer: any = shallow) => {
+  return renderer(<MinorButton {...props} />);
+};
+
+const isMinorSizesPadding: [SizeOptions, string][] = [
+  [
+    "small",
+    "var(--spacing000) var(--spacing100) var(--spacing000) var(--spacing100)",
+  ],
+  ["medium", "var(--spacing100)"],
+  ["large", "var(--spacing100)"],
+];
+
+interface Types {
+  background?: string;
+  borderColor?: string;
+  color?: string;
+}
+
+const isMinorColors: [ButtonTypes, Types][] = [
+  [
+    "primary",
+    {
+      background: "var(--colorsActionMinor500)",
+      borderColor: "var(--colorsActionMinorTransparent)",
+      color: "var(--colorsActionMinorYang100)",
+    },
+  ],
+  [
+    "secondary",
+    {
+      background: "transparent",
+      borderColor: "var(--colorsActionMinor500)",
+      color: "var(--colorsActionMinor500)",
+    },
+  ],
+  [
+    "tertiary",
+    {
+      background: "transparent",
+      borderColor: "transparent",
+      color: "var(--colorsActionMinor500)",
+    },
+  ],
+  [
+    "darkBackground",
+    {
+      borderColor: "transparent",
+    },
+  ],
+];
+
+describe("Button Minor", () => {
+  describe("Padding", () => {
+    it.each(isMinorSizesPadding)(
+      "when size is %s the padding is %s",
+      (size, padding) => {
+        const wrapper = mount(<MinorButton size={size}>Foo</MinorButton>);
+        assertStyleMatch({ padding }, wrapper);
+      }
+    );
+  });
+
+  describe("Colors", () => {
+    it.each(isMinorColors)("when buttonType is %s", (buttonType, styles) => {
+      const wrapper = mount(
+        <MinorButton buttonType={buttonType}>Foo</MinorButton>
+      );
+      assertStyleMatch({ ...styles }, wrapper);
+    });
+
+    describe("Colors for destructive and disabled", () => {
+      it("matches the style for the Button Minor destructive", () => {
+        const wrapper = render(
+          { children: "foo" },
+          TestRenderer.create
+        ).toJSON();
+        assertStyleMatch(
+          {
+            background: "transparent",
+            color: "var(--colorsActionMinor500)",
+          },
+          wrapper
+        );
+      });
+    });
```

Comment:
```suggestion
it("when destructive prop is passed, renders with destructive styling", () => {
    const wrapper = mount(<MinorButton destructive>foo</MinorButton>);
    assertStyleMatch(
      {
        background: "transparent",
        color: "var(--colorsSemanticNegative500)",
      },
      wrapper
    );
  });

  it("when disabled prop is passed, renders with disabled styling", () => {
    const wrapper = mount(<MinorButton disabled>foo</MinorButton>);
    assertStyleMatch(
      {
        background: "transparent",
        color: "var(--colorsActionMajorYin030)",
      },
      wrapper
    );
  });
```

I would use `mount` here rather than `TestRenderer`. That way you could also remove the custom `render` function as well. 

I think writing a `render` function would be more of an over-abstraction here with little benefits, since we have a small number of tests.